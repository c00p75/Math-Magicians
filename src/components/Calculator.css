
#calc-container > div > * {
  height: 70px;
}

#calc-container .col,
#calc-container .col-6 {
  border-top: 1px solid gainsboro;
  border-right: 1px solid gainsboro;
}

#calc-container .col:first-child,
#calc-container .col-6 {
  border-left: 1px solid gainsboro;
}

#calc-container .col:last-child {
  background-color: orange;
}

#calc-container .row:last-child {
  border-bottom: 1px solid gainsboro;
}

#calc-container button {
  width: 100%;
  border: 0;
  outline: none;
  cursor: pointer;
}

#calc-container button:active {
  filter: brightness(75%) !important;
}

#calc-container button:hover {
  filter: brightness(95%);
}

#calc-container #result {
  background-color: gray;
  font-weight: bold;
}

.calc-text {
  font-size: 2em;
  font-weight: bold;
  font-family: 'Poppins', sans-serif;
}
