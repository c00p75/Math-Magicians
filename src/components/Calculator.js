import { useState } from 'react';
import calculate from '../logic/calculate';

// Refactoring class component to function component which uses a React hook
const Calculator = () => {
  // initializing current state `initCalc` and function that updates the state `setCalculater`
  const [initCalc, setCalculator] = useState({ total: null, next: null, operation: null });

  // Defining function to handle click events and update state
  const handleClick = (event) => {
    // Changes initial calculator state by calling imported `calculate` function
    setCalculator(calculate(initCalc, event.target.innerHTML));
  };
  // Definfing HTML that should be rendered to the DOM
  return (
    <div className="container text-center mt-5" id="calc-container">
      <div className="row g-2">
        <div className="col d-flex justify-content-end align-items-center" id="result">
          {initCalc.next || initCalc.operation || initCalc.total || 0}
        </div>
      </div>

      <div className="row justify-content-center align-items-center g-2">
        <button onClick={handleClick} type="button" className="col">AC</button>
        <button onClick={handleClick} type="button" className="col">+/-</button>
        <button onClick={handleClick} type="button" className="col">%</button>
        <button onClick={handleClick} type="button" className="col">รท</button>
      </div>

      <div className="row justify-content-center align-items-center g-2">
        <button onClick={handleClick} type="button" className="col">7</button>
        <button onClick={handleClick} type="button" className="col">8</button>
        <button onClick={handleClick} type="button" className="col">9</button>
        <button onClick={handleClick} type="button" className="col">x</button>
      </div>

      <div className="row justify-content-center align-items-center g-2">
        <button onClick={handleClick} type="button" className="col">4</button>
        <button onClick={handleClick} type="button" className="col">5</button>
        <button onClick={handleClick} type="button" className="col">6</button>
        <button onClick={handleClick} type="button" className="col">-</button>
      </div>

      <div className="row justify-content-center align-items-center g-2">
        <button onClick={handleClick} type="button" className="col">1</button>
        <button onClick={handleClick} type="button" className="col">2</button>
        <button onClick={handleClick} type="button" className="col">3</button>
        <button onClick={handleClick} type="button" className="col">+</button>
      </div>

      <div className="row justify-content-center align-items-center g-2">
        <button onClick={handleClick} type="button" className="col-6">0</button>
        <button onClick={handleClick} type="button" className="col">.</button>
        <button onClick={handleClick} type="button" className="col">=</button>
      </div>
    </div>
  );
};

export default Calculator;
