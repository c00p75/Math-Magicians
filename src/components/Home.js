import React from 'react';

const Home = () => (
  <div className="pageContainer text-center">
    <h2 className="p-2" style={{ fontWeight: 'bold' }}>Welcome to our page</h2>
    <div>
      Math Magicians is a website for all fans of mathematics.
      Make simple calculations and view some of your favorite math quotes!
    </div>
  </div>
);

export default Home;
