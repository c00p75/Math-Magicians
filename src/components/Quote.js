import React from 'react';

const Quote = () => (
  <div className=" pageContainer text-center">
    <p style={{ fontFamily: 'Tangerine', fontSize: '2.5em' }}>
      &quot;Mathematics is not about numbers, equations, computations,
      or algorithms: it is about understanding.&quot;
    </p>
    <span>- Willian Paul Thurtson</span>
  </div>
);

export default Quote;
