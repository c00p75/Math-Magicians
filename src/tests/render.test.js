import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Home from '../components/Home';
import Calculator from '../components/Calculator';
import Navigation from '../components/Navbar';
import Quote from '../components/Quote';

it('Test if Home renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test if Calculator renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test if Quote renders correctly', () => {
  const tree = renderer.create(<Quote />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test if Home renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
