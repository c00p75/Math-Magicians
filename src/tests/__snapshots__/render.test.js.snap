// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Calculator renders correctly 1`] = `
<div
  className="container-fluid d-md-flex align-items-start justify-content-center"
>
  <div
    className="calc-text px-5 pt-5"
  >
    <p>
      Let's do some math!
    </p>
  </div>
  <div
    className="col container text-center pt-5"
    id="calc-container"
  >
    <div
      className="row g-2"
    >
      <div
        className="col d-flex justify-content-end align-items-center"
        id="result"
      >
        0
      </div>
    </div>
    <div
      className="row justify-content-center align-items-center g-2"
    >
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="row justify-content-center align-items-center g-2"
    >
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="row justify-content-center align-items-center g-2"
    >
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="row justify-content-center align-items-center g-2"
    >
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="row justify-content-center align-items-center g-2"
    >
      <button
        className="col-6"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="col"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Test if Home renders correctly 1`] = `
<div
  className="pageContainer text-center"
>
  <h2
    className="p-2"
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Welcome to our page
  </h2>
  <div>
    Math Magicians is a website for all fans of mathematics. Make simple calculations and view some of your favorite math quotes!
  </div>
</div>
`;

exports[`Test if Home renders correctly 2`] = `
<nav
  className="px-2 navBar d-flex justify-content-between align-items-center flex-wrap"
>
  <h1
    style={
      Object {
        "color": "aliceblue",
        "fontFamily": "Rubik Vinyl",
        "fontWeight": "bold",
      }
    }
  >
    Math Magicians
  </h1>
  <ul
    className="d-flex "
  >
    <li>
      <a
        aria-current="page"
        className="active"
        exact="true"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        exact="true"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        exact="true"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Test if Quote renders correctly 1`] = `
<div
  className=" pageContainer text-center"
>
  <p
    style={
      Object {
        "fontFamily": "Tangerine",
        "fontSize": "2.5em",
      }
    }
  >
    "Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding."
  </p>
  <span>
    - Willian Paul Thurtson
  </span>
</div>
`;
